cli/src/extension_api.ts

```diff
@@ -37,358 +39,380 @@ export class PhylumApi {
     );
   }
 
-  /// Get info about the logged in user.
-  ///
-  /// # Returns
-  ///
-  /// Object containing user information:
-  ///
-  /// ```
-  /// {
-  ///   email: "user@phylum.io",
-  ///   sub: "af8b5c32-9966-496a-e5ae-9ca9ceb43294",
-  ///   name: "John Doe",
-  ///   given_name: "John",
-  ///   family_name: "Doe",
-  ///   preferred_username: "JD",
-  ///   email_verified: true,
-  /// }
-  /// ```
-  static async getUserInfo(): object {
-    return await Deno.core.opAsync("get_user_info");
+  /**
+   * Get info about the logged in user.
+   *
+   * # Returns
+   *
+   * Object containing user information:
+   *
+   * ```
+   * {
+   *   email: "user@phylum.io",
+   *   sub: "af8b5c32-9966-496a-e5ae-9ca9ceb43294",
+   *   name: "John Doe",
+   *   given_name: "John",
+   *   family_name: "Doe",
+   *   preferred_username: "JD",
+   *   email_verified: true,
+   * }
+   * ```
+   */
+  static getUserInfo(): Promise<object> {
+    return Deno.core.opAsync("get_user_info");
   }
 
-  /// Get the current short-lived API access token.
-  static async getAccessToken(): string {
-    return await Deno.core.opAsync("get_access_token");
+  /** Get the current short-lived API access token. */
+  static getAccessToken(): Promise<string> {
+    return Deno.core.opAsync("get_access_token");
   }
 
-  /// Get the long-lived user refresh token.
-  static async getRefreshToken(): string {
-    return await Deno.core.opAsync("get_refresh_token");
+  /** Get the long-lived user refresh token. */
+  static getRefreshToken(): Promise<string> {
```

Comment:
What's the reason of having these not be async? Did you just change it because that's what Deno does?

I'd imagine it's more "compatible" since it doesn't require the "new" async keyword, but on the other hand I also feel like using `async` is probably the "recommended" way with the latest version of JavaScript?

Reply:
There are many technically valid ways to write this function:

1. Without `async` / `await`:
   ```
   static getRefreshToken(): Promise<string> {
     return Deno.core.opAsync("get_refresh_token");
   }
   ```
1. With `async`, but not `await`:
   ```
   static async getRefreshToken(): Promise<string> {
     return Deno.core.opAsync("get_refresh_token");
   }
   ```
1. With `async` / `await`:
   ```
   static async getRefreshToken(): Promise<string> {
     return await Deno.core.opAsync("get_refresh_token");
   }
   ```

You might be surprised (as I was) at a few details here:

1. They all use the `Promise<_>` return type. Type-check fails without it.
   ```
   TS1064 [ERROR]: The return type of an async function or method must be the global Promise<T> type. Did you mean to write 'Promise<string>'?
     static async getRefreshToken(): string {
   ```
1. It's valid to return an unawaited `Promise<_>` in an `async` function... This is a feature of JavaScript. An `async` function automatically wraps the return in a `Promise` if and only if it isn't already a `Promise`.

Since we can just return the same promise returned by `opAsync`, we don't need `await`. And if we don't need `await`, we don't need to mark the function as `async`.

Reply:
I was aware that you don't need to await it, but I wasn't aware that you need to declare the return type to be a Promise. I thought TypeScript did that automatically just like Rust. Sounds good then.